src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,310 @@
+
+            if (result === timeout) {
+                throw new Error("SRV record query timed out.");
+            }
+
+            const srvTargetHosts = result as string[];
+            // Shuffle the list of SRV target hosts
+            for (let i = srvTargetHosts.length - 1; i > 0; i--) {
+                const j = Math.floor(Math.random() * (i + 1));
+                [srvTargetHosts[i], srvTargetHosts[j]] = [srvTargetHosts[j], srvTargetHosts[i]];
+            }
+
+            const newDynamicClients: ConfigurationClientWrapper[] = [];
+            for (const host of srvTargetHosts) {
+                if (isValidEndpoint(host, this.#validDomain)) {
+                    const targetEndpoint = `https://${host}`;
+                    if (targetEndpoint.toLowerCase() === this.endpoint.toLowerCase()) {
+                        continue;
+                    }
+                    const client = this.#newConfigurationClient(targetEndpoint);
+                    newDynamicClients.push(new ConfigurationClientWrapper(targetEndpoint, client));
+                }
+            }
+
+            this.#dynamicClients = newDynamicClients;
+            this.#lastFallbackClientRefreshTime = Date.now();
+        } catch (err) {
+            console.warn(`Fail to build fallback clients, ${err.message}`);
+        }
+    }
+
+    #newConfigurationClient(endpoint) {
+        if (this.#credential) {
+            return new AppConfigurationClient(endpoint, this.#credential, this.#clientOptions);
+        }
+
+        const connectionStr = buildConnectionString(endpoint, this.#secret, this.#id);
+        return new AppConfigurationClient(connectionStr, this.#clientOptions);
+    }
+
+    #isFallbackClientDiscoveryDue(dateTime) {
+        return dateTime >= this.#lastFallbackClientRefreshAttempt + MINIMAL_CLIENT_REFRESH_INTERVAL
+            && (!this.#dynamicClients
+                || this.#dynamicClients.every(client => dateTime < client.backoffEndTime)
+                || dateTime >= this.#lastFallbackClientRefreshTime + FALLBACK_CLIENT_REFRESH_EXPIRE_INTERVAL);
+    }
+}
+
+/**
+ * Query SRV records and return target hosts.
+ */
+async function querySrvTargetHost(host: string): Promise<string[]> {
+    const results: string[] = [];
+    let dns;
+
+    if (typeof global !== "undefined" && global.dns) {
+        dns = global.dns;
+    } else {
+        throw new Error("Failover is not supported in the current environment.");
+    }
+
+    try {
+        // Look up SRV records for the origin host
+        const originRecords = await dns.resolveSrv(`${TCP_ORIGIN_KEY_NAME}.${host}`);
+        if (originRecords.length === 0) {
+            return results;
+        }
+
+        // Add the first origin record to results
+        const originHost = originRecords[0].name;
+        results.push(originHost);
+
+        // Look up SRV records for alternate hosts
+        let index = 0;
+        let moreAltRecordsExist = true;
+        while (moreAltRecordsExist) {
```

Comment:
```suggestion
        // eslint-disable-next-line no-constant-condition
        while (true) {
```

And remove `moreAltRecordsExist`